osquery/tables/system/linux/block_devices.cpp

```diff
@@ -9,20 +9,61 @@
  */
 
 #include <boost/algorithm/string.hpp>
+#include <boost/algorithm/string/predicate.hpp>
 
 #include <blkid/blkid.h>
 #include <libudev.h>
+#include <linux/kdev_t.h>
 #include <unistd.h>
 
 #include <osquery/core.h>
 #include <osquery/filesystem.h>
 #include <osquery/logger.h>
 #include <osquery/tables.h>
 
+extern "C" {
+#include <lvm2app.h>
+}
+
 namespace osquery {
 namespace tables {
 
-static void getBlockDevice(struct udev_device *dev, QueryData &results) {
+void populatePVChildren(lvm_t lvm,
+                        const std::string& devname,
+                        const std::string& pvid,
+                        std::map<std::string, std::string>& lvm_lv2pv) {
+  const auto pvid_copy = boost::erase_all_copy(pvid, "-");
+  const char* vg_name = lvm_vgname_from_pvid(lvm, pvid_copy.c_str());
+  if (vg_name == nullptr) {
+    return;
+  }
+
+  vg_t vg = lvm_vg_open(lvm, vg_name, "r", 0);
+  if (vg == nullptr) {
+    return;
+  }
+  lv_list_t* lv = nullptr;
+  struct dm_list* lvs = lvm_vg_list_lvs(vg);
+  if (lvs != nullptr) {
+    dm_list_iterate_items(lv, lvs) {
+      struct lvm_property_value kernel_major = lvm_lv_get_property(
+                                    lv->lv, "lv_kernel_major"),
+                                kernel_minor = lvm_lv_get_property(
+                                    lv->lv, "lv_kernel_minor");
+      const uint64_t major = kernel_major.value.integer,
+                     minor = kernel_minor.value.integer;
+      const char* child_devname = blkid_devno_to_devname(MKDEV(major, minor));
```

Comment:
Assuming this is guaranteed to be null terminated given it's implicit cast to std::string below. ;)